src/clr-core/common/utils/aria-live.service.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { AriaLivePoliteness } from '../enums/aria-live-politeness';
+import { applyMixins } from '../mixins/apply-mixins';
+import { UniqueId } from '../mixins/unique-id';
+
+/**
+ * Time in milliseconds before inserting the content into the container
+ */
+export const ARIA_LIVE_TICK: number = 100;
+export class CwcAriaLiveService {
+  protected previousTimeout: ReturnType<typeof setTimeout>;
+  protected ariaLiveElement: HTMLElement | null;
+  document: Document;
+
+  readonly id = `clr-aria-live-element-${this._uniqueId}`;
+
+  constructor(document: Document) {
+    this.document = document;
+  }
+
+  /**
+   * Append text content inside the AriaLive Container. This method will check if the
+   * DOM Element is existing if not it will create one for us and the will apply the text.
+   *
+   * ```typescript
+   * this.ariaLiveService.announce(this.el.nativeElement);
+   * // or
+   * this.ariaLiveService.announce('Message to announce to SR');
+   * ```
+   *
+   * @remark
+   * When second argument is `AriaLivePoliteness.off` we won't create aria container or update it.
+   * The reason for that is that we don't want to do additional work if the SR will ignore it.
+   *
+   * @param message - This could be simple string or HTMLElement
+   * @param politeness - 'polite', 'assertive' or 'off'
+   */
+  announce(message: string | HTMLElement, politeness: AriaLivePoliteness = AriaLivePoliteness.polite): void {
+    if (!message || politeness === AriaLivePoliteness.off) {
+      return;
+    }
+
+    if (typeof message !== 'string' && this.isBrowser() && message.textContent) {
+      message = message.textContent;
+    }
+
+    // when there is no message do NOTHING!
+    if (!message) {
```

Comment:
I see.

It's likely that the flow of this code could be improved. My first instinct was to remove the check on ln 53. Reviewing this again, I can see that would be bad.

But I could also see a dev that is unfamiliar with this code doing that and even imagine it passing review.

So that tells me something about the code itself.

I think we could simplify this guard and make it clearer that this isn't an unnecessary line of code.

Reply:
i totally agree with that, the comments need improvements. Not sure how could he simplify the code .. maybe better naming could give a hint what is going on

Reply:
But...

...if that is true then `announce()` only adds text to the live element when there is no live element.

So where is the reuse?

Can I not call `announce()` more than once on the same aria-live element?

Because the code says that is the case...